src/dyn_subgrid/dynFATESLandUseChangeMod.F90

```diff
@@ -0,0 +1,194 @@
+module dynFATESLandUseChangeMod
+
+#include "shr_assert.h"
+
+  !---------------------------------------------------------------------------
+  ! !DESCRIPTION:
+  ! Handle reading of the land use harmonization (LUH2) dataset
+
+  ! !USES:
+  use shr_kind_mod          , only : r8 => shr_kind_r8
+  use shr_log_mod           , only : errMsg => shr_log_errMsg
+  use decompMod             , only : bounds_type, BOUNDS_LEVEL_PROC
+  use abortutils            , only : endrun
+  use dynFileMod            , only : dyn_file_type
+  use dynVarTimeUninterpMod , only : dyn_var_time_uninterp_type
+  use clm_varcon            , only : grlnd
+  use clm_varctl            , only : iulog
+
+  implicit none
+
+  private
+
+  real(r8), allocatable, public :: landuse_transitions(:,:)
+  real(r8), allocatable, public :: landuse_states(:,:)
+
+  integer, public, parameter    :: num_landuse_transition_vars = 108
+  integer, public, parameter    :: num_landuse_state_vars = 12
+
+  type(dyn_file_type), target   :: dynFatesLandUse_file
+
+  ! Land use name arrays
+  character(len=5), public, parameter  :: landuse_state_varnames(num_landuse_state_vars) = &
+                    [character(len=5)  :: 'primf','primn','secdf','secdn','pastr','range', &
+                                          'urban','c3ann','c4ann','c3per','c4per','c3nfx']
+
+  character(len=14), public, parameter :: landuse_transition_varnames(num_landuse_transition_vars) = &
+                    [character(len=14) :: 'primf_to_secdn','primf_to_pastr','primf_to_range','primf_to_urban', &
+                                          'primf_to_c3ann','primf_to_c4ann','primf_to_c3per','primf_to_c4per','primf_to_c3nfx', &
```

Comment:
This list is so long that I wonder if there might be another way to manage it? Maybe in a different file? Or maybe you process all the variables and use the variable list from the file?

But, that might only matter if we think this might change in the future? How likely is anything here to change?

Reply:
This is unlikely to change unless the [LUH](https://luh.umd.edu/) update for LUH3 changes the naming or number (longer term consideration).  Thinking about how to manage these sort of lists generally is not a bad idea though.  Maybe I can make an issue/discussion for it?

Reply:
@ekluzek I have an alternate method of approaching this on a different branch (that I haven't tested yet), but I'm wondering what you think: 8434cbd2f48e6f03dc6c24ebb45f7dd5292a50a8

Reply:
Per our discussion today, @ekluzek and I agree to move forward with the long list of parameters as is currently.  I will have a forthcoming refactor of the LUH2 data tool (to be included with https://github.com/NGEET/fates/pull/1107) in which I will explore incorporating the metadata approached discussed here: https://github.com/glemieux/fates/issues/30#issuecomment-1703210904